### PR TITLE
Update expected output of some test files when parallel query is enforced.

### PR DESCRIPTION
**Pull request by skumawat2025**
### Description

Update expected output of some test files in case of parallel query as query plan changed. Also unblock these tests from parallel query jdbc ignore file. 

Reason for expected output diff in case of parallel query mode - 

```
# Output rows ORDER changes.
BABEL-2514 - order changes because of UNION
BABEL-2998 - order changes because of UNION 
BABEL-328-vu-verify - order changes because of JOIN
TestSQLQueries - order changes because of JOIN

BABEL-1444 - We don't share state of the GUC "role" with parallel worker which will result in current_setting('role') to return none

# Query plan change
BABEL-1363
BABEL-2843
BABEL-2844
BABEL-2902
BABEL-2903
BABEL-3248
BABEL-3295
BABEL-3513-vu-prepare
BABEL-3513-vu-verify
BABEL-4281
babel_collection
binary-index-vu-verify
cast_eliminate-vu-verify
babel_index_nulls_order-vu-verify
BABEL-4264

```

Signed-off-by: Sandeep Kumawat <skumwt@amazon.com>
### Issues Resolved
BABEL-4643

### Test Scenarios Covered ###
* **Use case based -**

Test files are already there moved the updated output inside parallel_expected/* folder.

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/parallel_query/BABEL-1363.out [hunk 1]
```diff
@@ -0,0 +1,64 @@
+-- NOTE: this test should not run in parallel due to DB config change.
```
Comment (Deepesh125) [inline test/JDBC/expected/parallel_query/BABEL-1363.out]: As per this comment, lets ignore this test in parallel query testing

File: test/JDBC/expected/parallel_query/TestSQLQueries.out [hunk 2]
```diff
@@ -0,0 +1,839 @@
+#1 CREATE TABLE with primary and unique keyword
```
Comment (Deepesh125) [inline test/JDBC/expected/parallel_query/TestSQLQueries.out:1]: why is this file required to be different in parallel query tesing?

File: test/JDBC/expected/parallel_query/binary-index-vu-prepare.out [hunk 3]
```diff
@@ -0,0 +1,68 @@
+create table tab_binary (a binary(4), b int)
```
Comment (Deepesh125) [inline test/JDBC/expected/parallel_query/binary-index-vu-prepare.out]: why diff expected file for parallel query?

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: test/JDBC/expected/parallel_query/TestSQLQueries.out (hunk 2 above)
Comment (skumawat2025) [reply test/JDBC/expected/parallel_query/TestSQLQueries.out:1]: order of output rows changes because of JOINs in case of parallel query. 

```
--- /workplace/skumwt/babelfish_extensions/test/JDBC/./expected/TestSQLQueries.out	2023-12-04 12:55:48.254859720 +0000
+++ /workplace/skumwt/babelfish_extensions/test/JDBC/./output/TestSQLQueries.out	2023-12-25 09:30:45.118615042 +0000
@@ -596,25 +596,25 @@
 SELECT temp1.i, temp1.j, temp1.t, temp2.k FROM temp1 INNER JOIN temp2 ON temp1.i=temp2.i;
 ~~START~~
 int#!#int#!#text#!#int
-0#!#<NULL>#!#zero#!#<NULL>
 1#!#4#!#one#!#-1
 2#!#3#!#two#!#2
-2#!#3#!#two#!#4
 3#!#2#!#three#!#-3
+2#!#3#!#two#!#4
 5#!#0#!#five#!#-5
 5#!#0#!#five#!#-5
+0#!#<NULL>#!#zero#!#<NULL>
 ~~END~~
 
 SELECT temp1.i, temp1.j, temp1.t, temp2.k FROM temp1 JOIN temp2 ON temp1.i=temp2.i;
 ~~START~~
 int#!#int#!#text#!#int
-0#!#<NULL>#!#zero#!#<NULL>
 1#!#4#!#one#!#-1
 2#!#3#!#two#!#2
-2#!#3#!#two#!#4
 3#!#2#!#three#!#-3
+2#!#3#!#two#!#4
 5#!#0#!#five#!#-5
 5#!#0#!#five#!#-5
+0#!#<NULL>#!#zero#!#<NULL>
 ~~END~~
 
 
@@ -622,7 +622,6 @@
 SELECT * FROM temp1 LEFT OUTER JOIN temp2 ON temp1.i=temp2.k;
 ~~START~~
 int#!#int#!#text#!#int#!#int
-0#!#<NULL>#!#zero#!#<NULL>#!#0
 1#!#4#!#one#!#<NULL>#!#<NULL>
 2#!#3#!#two#!#2#!#2
 3#!#2#!#three#!#<NULL>#!#<NULL>
@@ -631,6 +630,7 @@
 6#!#6#!#six#!#<NULL>#!#<NULL>
 7#!#7#!#seven#!#<NULL>#!#<NULL>
 8#!#8#!#eight#!#<NULL>#!#<NULL>
+0#!#<NULL>#!#zero#!#<NULL>#!#0
 <NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
 <NULL>#!#0#!#zero#!#<NULL>#!#<NULL>
 ~~END~~
@@ -640,13 +640,13 @@
 SELECT * FROM temp1 RIGHT OUTER JOIN temp2 ON temp1.i=temp2.i;
 ~~START~~
 int#!#int#!#text#!#int#!#int
-0#!#<NULL>#!#zero#!#0#!#<NULL>
 1#!#4#!#one#!#1#!#-1
 2#!#3#!#two#!#2#!#2
-2#!#3#!#two#!#2#!#4
 3#!#2#!#three#!#3#!#-3
+2#!#3#!#two#!#2#!#4
 5#!#0#!#five#!#5#!#-5
 5#!#0#!#five#!#5#!#-5
+0#!#<NULL>#!#zero#!#0#!#<NULL>
 <NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#<NULL>
 <NULL>#!#<NULL>#!#<NULL>#!#<NULL>#!#0
 ~~END~~

```

File: test/JDBC/expected/parallel_query/binary-index-vu-prepare.out (hunk 3 above)
Comment (skumawat2025) [reply test/JDBC/expected/parallel_query/binary-index-vu-prepare.out]: Yes, this file shouldn't have a output diff for parallel query. removing it. thanks.

**Review by Deepesh125 — APPROVED** [review]
